autoload/neomake/makers/ft/javascript.vim

```diff
@@ -41,11 +41,17 @@ function! neomake#makers#ft#javascript#standard()
 endfunction
 
 function! neomake#makers#ft#javascript#flow()
-    " Replace "\n" by space.
-    let mapexpr = 'substitute(v:val, "\\\\n", " ", "g")'
+    " Multi-line Vim
+    " \ 'errorformat': '%CFile %m%.%#,%AFile "%f"\, line %l\, characters %c-%.%#,%+G%m,%Z%m,%-G%.%#'
+
+    " Single-line Vim - Long
+    " 'errorformat': '%CFile %m%.%#,%AFile "%f"\, line %l\, characters %c-%.%#,%C%m,%Z%m,%-G%.%#'
+
+    " Single-line Vim - Short
+    " \ 'errorformat': '%CFile %.%#,%AFile "%f"\, line %l\, characters %c-%.%#,%C%m,%Z%m,%-G%.%#'
+
     return {
-        \ 'args': ['check', '--one-line'],
-        \ 'errorformat': '%f:%l:%c\,%n: %m',
-        \ 'mapexpr': mapexpr,
+        \ 'args': ['--from=vim'],
+        \ 'errorformat': '%CFile %.%#,%AFile "%f"\, line %l\, characters %c-%.%#,%C%m,%Z%m'
```

Comment:
I noticed that this errorformat isn't mentioned in the comments above. How does it differ from those? Might want to add it above so that future readers know which is selected and why it was deemed to hold the advantage over the other options. Or, if you are confident in your choice, you could simply remove the comments above :)